community/server/src/main/java/org/neo4j/server/rest/transactional/Neo4jJsonCodec.java

```diff
@@ -28,6 +28,7 @@
 import org.codehaus.jackson.map.ObjectMapper;
 import org.codehaus.jackson.map.SerializationConfig;
 
+import org.neo4j.cypher.internal.compiler.v3_0.commands.expressions.ActsAsMap;
```

Comment:
I don't like that `server` depends directly on types defined in `3.0`. This means that when we forward merge this `server` needs to check both for a `3.0` and a `3.1` `ActsAsMap`, and it will only get worse in future versions. I also believe we should avoid using `scala` from `java`. 

I think we need a `GeographicPoint` that lives in the public API and convert to this in the compatibility layer.